### ISSUE TITLE
curly rule or docs bug

### ISSUE THREAD

**Issue by AuthorProxy** [eslint/eslint#11162]
Version: `"eslint": "^4.19.1"`
Code sample - any oneliner - `if (smth) return null;`

Documentation: `https://eslint.org/docs/rules/curly` says that `all` is default with relevant code samples.

my js config:
`curly: 'error',` - not works on single line statement without brackets
`curly: ['error', 'all'],` - works as expected

**Comment by platinumazure** [comment]
Hi @AuthorProxy, thanks for the issue.

What do you mean by "works", here? Can you please be more explicit about what you see and what you are expecting to see? Thanks!

**Comment by AuthorProxy** [comment]
1. "eslint": "^5.12.1", "babel-eslint": "^10.0.1"
2. 
Under 'works' I understand the `correct` work of rule, if it do what it should do.
Documentation says, that `"all" is default option` and gives sample of incorrect code `if (foo) foo++;`. But really it doesn't work, no errors are shown with this configuration: `curly: 'error'`, but if I change it to `curly: ['error', 'all'],` - it works as expected.

So default `"all"` option is not work until I specify it explicitly at config.

Code sample:
```
    const expression = !!'1';
    if (expression) console.log('1');
```

3. Current output: no errors
4. Expected output:  error eslint:curly        Expected { after 'if' condition.

**Comment by aladdin-add** [comment]
however, I was not able to repro it using the default parser: [online demo](https://eslint.org/demo/#eyJ0ZXh0IjoiLyplc2xpbnQgY3VybHk6IFsnZXJyb3InXSAqL1xyXG5pZiAoZm9vKSBmb28rKzsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjksInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnsianN4Ijp0cnVlfX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

**Comment by AuthorProxy** [comment]
Check this sample: https://github.com/AuthorProxy/eslint_curly_error_sample
Can reproduce it after importing airbnb config
